posts/2025-02-17_theres_a_pharmaverse_package_for_that/managing-clinical-trial-data.qmd

```diff
@@ -0,0 +1,323 @@
+data("dm", package = "pharmaversesdtm")
+data("ex", package = "pharmaversesdtm")
+data("ds", package = "pharmaversesdtm")
+```
+
+ADSL is typically **built from the DM dataset**, removing unnecessary columns and adding treatment variables in one step:
+
+``` r
+adsl <- dm %>%
+  select(-DOMAIN) %>%
+  mutate(
+    TRT01P = ARM,
+    TRT01A = ACTARM
+  )
+```
+
+#### **Step 2: Derive Treatment Variables**
+
+Using {admiral}, we extract and standardize treatment dates from the EX dataset:
+
+``` r
+ex_ext <- ex %>%
+  filter(!is.na(USUBJID)) %>%
+  derive_vars_dt(
+    dtc = EXSTDTC,
+    new_vars_prefix = "EXST"
+  ) %>%
+  derive_vars_dt(
+    dtc = EXENDTC,
+    new_vars_prefix = "EXEN"
+  )
+```
+
+#### Then merge these dates into ADSL:
+
+``` r
+adsl <- adsl %>%
+  derive_vars_merged(
+    dataset_add = ex_ext,
+    filter_add = EXDOSE > 0 | (EXDOSE == 0 & str_detect(EXTRT, "PLACEBO")),
+    new_vars = exprs(TRTSDT = EXSTDT, TRTEDT = EXENDT),
+    by_vars = exprs(STUDYID, USUBJID)
+  )
+```
+
+#### **Step 3: Derive End of Study (EOS) Status**
+
+The disposition dataset (DS) is used to determine when a patient exited the study:
+
+``` r
+ds_ext <- ds %>%
+  filter(!is.na(DSSTDTC)) %>%
+  derive_vars_dt(
+    dtc = DSSTDTC,
+    new_vars_prefix = "DSST"
+  )
+
+adsl <- adsl %>%
+  derive_vars_merged(
+    dataset_add = ds_ext,
+    by_vars = exprs(STUDYID, USUBJID),
+    new_vars = exprs(EOSDT = DSSTDT),
+    filter_add = DSCAT == "DISPOSITION EVENT" & DSDECOD != "SCREEN FAILURE"
+  )
+```
+
+#### **Step 4: Assign Population Flags**
+
+For safety population (SAFFL), we check if the patient received a treatment dose:
+
+``` r
+adsl <- adsl %>%
+  derive_var_merged_exist_flag(
+    dataset_add = ex,
+    by_vars = exprs(STUDYID, USUBJID),
+    new_var = SAFFL,
+    condition = EXDOSE > 0 | str_detect(EXTRT, "PLACEBO")
+  )
+```
+
+#### **Step 5: Generate and Save Results**
+
+Finally, we print the results and save them to a CSV file:
+
+``` r
+tryCatch({
+  print("ADSL creation complete. Showing first few rows:")
+  print(head(adsl))
+  
+  print("\nSummary of ADSL variables:")
+  print(summary(adsl))
+  
+  # Save to a CSV file
+  write.csv(adsl, "adsl_output.csv", row.names = FALSE)
+  print("\nADSL has been saved to 'adsl_output.csv'")
+}, error = function(e) {
+  print(paste("An error occurred:", e$message))
+})
+```
+
+![](images/result1.png)
+
+![](images/result2.png)\
+
+#### **More Details on ADSL Creation**
+
+This is just a **high-level example**; the full process includes deriving death variables, grouping populations, and applying labels. For a deeper dive, check out the [ADSL Implementation Guide](https://cran.r-project.org/web/packages/admiral/vignettes/adsl.html).
+
+![](images/adsl.png)
+
+## **Who Are the Key Players in Pharmaverse, and Do You Need to Use All Packages?**
+
+[![Source: Getting Started with tidyverse in R](images/tidyverse-packages.png)](https://www.storybench.org/getting-started-with-tidyverse-in-r/)
```

Comment:
Question: does this image make sense here? the section is about pharmaverse but the image is tidyverse